pilot/pkg/serviceregistry/serviceentry/util.go

```diff
@@ -49,3 +50,18 @@ func difference(old, curr map[types.NamespacedName]*config.Config) []types.Names
 
 	return out
 }
+
+// difference returns the elements in `old` that aren't in `curr`.
+func workloadInstanceDiff(old, curr []*model.ServiceInstance) []*model.ServiceInstance {
+	mb := make(map[*model.ServiceInstance]struct{}, len(curr))
+	for _, x := range curr {
+		mb[x] = struct{}{}
+	}
+	var diff []*model.ServiceInstance
+	for _, x := range old {
+		if _, found := mb[x]; !found {
```

Comment:
comparing pointer doesn't seem entirely safe. Are we sure we never have two representations of the same instance but with a different pointer?

Reply:
It works in this specific scenario but I'm not sure about all use cases.

Do you think it'd better to use a Endpoint specific key to get the diff ? Something like `x.Endpoint.Labels.String() + x.Endpoint.Address + ":" + strconv.Itoa(int(x.Endpoint.EndpointPort))`

That'll make the diff more generic

Reply:
+1 with John, we should not compare pointer here.

Reply:
Ended up removing the comparison method altogether because the only place we use it, is where we knew for sure that the labels don't match. So the diff would always return the currentPod.